src/box/memtx_engine.c

```diff
@@ -159,8 +159,16 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 	/* Process existing snapshot */
 	say_info("recovery start");
 	int64_t signature = vclock_sum(vclock);
-	const char *filename = xdir_format_filename(&memtx->snap_dir,
+	char filename[PATH_MAX];
+	const char *name = xdir_format_filename(&memtx->snap_dir,
 						    signature, NONE);
+	/*
+	 * we need to save name in local variable because
+	 * xdir_format_filename allocate memory in static buffer
+	 * which will be overwritten later
+	 */
+	strncpy(filename, name, PATH_MAX - 1);
+	filename[PATH_MAX - 1] = '\0';
```

Comment:
As I understand, it is separate bug which is not connected with purpose of this patch. So I suggest you to move this change in a separate patch and also verify other xdir_format_filename() usages. Nit: start comment with a capital letter and end it with a dot.